test/Core.Test/OrganizationFeatures/AuthorizationHandlers/CollectionAccessAuthorizationHandlerBaseTests.cs

```diff
@@ -0,0 +1,228 @@
+            new ClaimsPrincipal(),
+            collectionUsers
+        );
+
+        sutProvider.GetDependency<ICurrentContext>().UserId.Returns(actingUserId);
+        sutProvider.GetDependency<ICollectionRepository>().GetManyByManyIdsAsync(Arg.Any<IEnumerable<Guid>>()).Returns(collections);
+
+        await sutProvider.Sut.HandleAsync(context);
+        Assert.True(context.HasFailed);
+        await sutProvider.GetDependency<ICollectionRepository>().ReceivedWithAnyArgs().GetManyByManyIdsAsync(default);
+        await sutProvider.GetDependency<ICurrentContext>().DidNotReceiveWithAnyArgs()
+            .OrganizationMembershipAsync(default, default);
+    }
+
+    [Theory, BitAutoData, CollectionCustomization]
+    public async Task CanManageCollectionAccessAsync_MissingOrgMembership_Failure(
+        SutProvider<CollectionUserAuthorizationHandler> sutProvider,
+        ICollection<Collection> collections,
+        ICollection<CollectionUser> collectionUsers,
+        ICollection<CurrentContentOrganization> organizations)
+    {
+        var actingUserId = Guid.NewGuid();
+
+        var context = new AuthorizationHandlerContext(
+            new[] { CollectionUserOperation.Create },
+            new ClaimsPrincipal(),
+            collectionUsers
+        );
+
+        // Simulate a collection that belongs to an unknown organization
+        collections.First().OrganizationId = Guid.NewGuid();
+
+        sutProvider.GetDependency<ICurrentContext>().UserId.Returns(actingUserId);
+        sutProvider.GetDependency<ICollectionRepository>().GetManyByManyIdsAsync(Arg.Any<IEnumerable<Guid>>()).Returns(collections);
+        sutProvider.GetDependency<ICurrentContext>().OrganizationMembershipAsync(Arg.Any<IOrganizationUserRepository>(), actingUserId).Returns(organizations);
+
+        await sutProvider.Sut.HandleAsync(context);
+        Assert.True(context.HasFailed);
+        await sutProvider.GetDependency<ICollectionRepository>().ReceivedWithAnyArgs().GetManyByManyIdsAsync(default);
+        await sutProvider.GetDependency<ICurrentContext>().ReceivedWithAnyArgs().OrganizationMembershipAsync(default, default);
+        await sutProvider.GetDependency<ICollectionRepository>().DidNotReceiveWithAnyArgs()
+            .GetManyByUserIdAsync(default);
+    }
+
+    [Theory, BitAutoData, CollectionCustomization]
+    public async Task CanManageCollectionAccessAsync_MissingManageCollectionPermission_Failure(
+        SutProvider<CollectionUserAuthorizationHandler> sutProvider,
+        ICollection<Collection> collections,
+        ICollection<CollectionUser> collectionUsers,
+        ICollection<CollectionDetails> collectionDetails,
+        ICollection<CurrentContentOrganization> organizations)
+    {
+        var actingUserId = Guid.NewGuid();
+
+        // Simulate not having manage collection permission
+        foreach (var collectionDetail in collectionDetails)
+        {
+            collectionDetail.Manage = false;
+        }
```

Comment:
Suggestion: do this for just 1 `collectionDetail`, being the minimum required to get a `Context.Fail()` result.